### PR TITLE
feat: bridge modal UI and network switch for mainnet and testnet

### PR DESCRIPTION
**Pull request by 0xthi**
-Added both testnet and mainnet switch
-Bridge Modal completed for base sepolia
-Fixed code redundancy issues
-Networks can be added automatically if not vailable in metamask
-Deposit and Withdraw components are disabled for wrong networks

### CODE REVIEW

File: src/app/trade/page.tsx [hunk 1]
```diff
@@ -52,13 +55,7 @@ export default function TradePage() {
...
                 <div className='px-3 pt-[53px] pb-28 bg-[#0F121A]'>
                     <Separator orientation='horizontal' className='w-full h-px' />
-                    <Button
-                        onClick={() => setisDepositModalOpen(true)}
-                        variant={"outline"}
-                        className='border-[#2DA8CE] border hover:bg-transparent hover:scale-105 transition-all ease-in-out rounded-[8px] text-[#2DA8CE] w-full bg-transparent'
-                    >
-                        Deposit
-                    </Button>
+                    <DepositButtonWrapper variant="trade" />
```
Comment (Prithvi101) [inline src/app/trade/page.tsx]: you can remove it from here and add it to DailogWrapper component

File: src/components/modals/BridgeComponent.tsx [hunk 2]
```diff
@@ -1,158 +1,301 @@
...
-                </button>
-              </div>
-              {quote && (
-                <div className="mt-4 p-4 bg-gray-800 rounded shadow-lg">
-                  <pre className="text-gray-200 overflow-x-auto whitespace-pre-wrap break-words">
-                    {JSON.stringify(quote, replaceBigInt, 2)}
-                  </pre>
+
+    return (
+        <div className='flex flex-col gap-6'>
```
Comment (Prithvi101) [inline src/components/modals/BridgeComponent.tsx]: You can use BaseModal Component and take reference from addfundsmodal

File: src/components/modals/DepositButtonWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    const [, setIsDepositModalOpen] = useAtom(depositModalAtom)
+    const { isBridgeNetwork } = useNetworkState()
+
+    const buttonStyles = {
+        default: 'px-4 lg:px-10 bg-[#212124] border-none text-sm font-medium',
+        trade: 'border-[#2DA8CE] border hover:bg-transparent hover:scale-105 transition-all ease-in-out rounded-[8px] text-[#2DA8CE] w-full bg-transparent'
+    }
+
+    return (
+        <SignInRedirectorLayout>
```
Comment (Prithvi101) [inline src/components/modals/DepositButtonWrapper.tsx]: you can use dailogue wrapper instead

**Review by Prithvi101 — CHANGES_REQUESTED** [review]

File: src/components/modals/DepositButtonWrapper.tsx (hunk 3 above)
Comment (0xthi) [reply src/components/modals/DepositButtonWrapper.tsx]: I've made the above changes. Could you review it pls

File: src/components/layouts/DialogsWrapper.tsx [hunk 4]
```diff
@@ -1,35 +1,53 @@
...
+
     return (
         <div>
             {isDepositModalOpen && <DepositModal />}
             {isWithdrawModalOpen && <WithdrawModal />}
             {isSettingsModalOpen && <SettingsModal />}
             {isAddFundsModalOpen && <AddFundsModal />}
 
-            {children}
+            {children || (
```
Comment (Prithvi101) [inline src/components/layouts/DialogsWrapper.tsx]: not understanding the purpose of this  why are we making some optional component with children

File: src/app/portfolio/page.tsx [hunk 5]
```diff
@@ -113,15 +112,7 @@ const PortfolioPage = () => {
...
-                                    <SignInRedirectorLayout>
-                                        <Button
-                                            variant={"outline"}
-                                            className='px-4 lg:px-10 bg-[#212124] border-none text-sm font-medium'
-                                            onClick={() => setIsDepositModalOpen(true)}
-                                        >
-                                            Deposit
-                                        </Button>
-                                    </SignInRedirectorLayout>
+                                    <DialogsWrapper variant="default" />
```
Comment (Prithvi101) [inline src/app/portfolio/page.tsx]: why re we calling dailogue wrapper here

File: src/app/trade/page.tsx [hunk 6]
```diff
@@ -52,13 +55,7 @@ export default function TradePage() {
...
                 <div className='px-3 pt-[53px] pb-28 bg-[#0F121A]'>
                     <Separator orientation='horizontal' className='w-full h-px' />
-                    <Button
-                        onClick={() => setisDepositModalOpen(true)}
-                        variant={"outline"}
-                        className='border-[#2DA8CE] border hover:bg-transparent hover:scale-105 transition-all ease-in-out rounded-[8px] text-[#2DA8CE] w-full bg-transparent'
-                    >
-                        Deposit
-                    </Button>
+                    <DialogsWrapper variant="trade" />
```
Comment (Prithvi101) [inline src/app/trade/page.tsx]: same for here

**Review by Prithvi101 — CHANGES_REQUESTED** [review]

**Review by Prithvi101 — APPROVED** [review]

**Pull request merged**